src/index.js

```diff
@@ -45,6 +46,8 @@ const Multiaddr = withIs.proto(function (addr) {
   } else {
     throw new Error('addr must be a string, Buffer, or another Multiaddr')
   }
+
+  this.resolvers = new Map()
```

Comment:
This map should be static, and not on each instance. I should be able to set this once in the module where resolution is taking place. This can cause issues if multiple version of multiaddr are in the dep tree, but it's fine when the same module is setting and using the resolvers. Setting this on every multiaddr is not a great experience.